runtime/client_http_request.go

```diff
@@ -78,6 +78,32 @@ func (req *ClientHTTPRequest) start() {
 	req.metrics.Sent.Inc(1)
 }
 
+// CheckHeaders verifies that the outbound request contaisn required headers
+func (req *ClientHTTPRequest) CheckHeaders(expected []string) error {
+	if req.httpReq == nil {
+		/* coverage ignore next line */
+		panic("must call `req.WriteJSON()` before `req.CheckHeaders()`")
+	}
+
+	actualHeaders := req.httpReq.Header
+
+	for _, headerName := range expected {
+		// TODO: case sensitivity ?
+		headerValue := actualHeaders.Get(headerName)
+		if headerValue == "" {
+			req.Logger.Warn("Got outbound request without mandatory header",
```

Comment:
Should this be logged as an Error, since we failed the required headers check?  Or are you logging such client errors as Warnings?

Reply:
an `Error` is a pager duty offence. Do you want to get woken up for it ?

Reply:
An error is a bug; aka broken code. In theory this can be an `Error()` but it's possible for a user of edge-gateway to configure "mandatory" headers on the client but no mandatory headers on the endpoint and no header proxying.